packages/server/src/adapters/ws.ts

```diff
@@ -313,7 +324,7 @@ export function getWSConnectionHandler<TRouter extends AnyRouter>(
               break;
             }
 
-            const result: TRPCResultMessage<unknown>['result'] = {
+            result = {
```

Comment:
This looks unnecessary - if it's in the loop instead it should be GC'd, no?

Reply:
I see your comment _"We need those declarations outside the loop for garbage collection reasons."_ 


can you explain why?

it's within a closure... it should be freed automatically?

Reply:
Unfortunately, no. I thought so, too. Until I had those test cases proving me wrong.

I mentioned this in https://github.com/nodejs/node/issues/55585. It seems like a `const xyz = ...` inside a loop is not freed up at the end of the closure, but only when the same statement is reached again. I assume this to be some v8 optimization or bug. Maybe the node devs will tell me within that issue I opened on their side.

Reply:
Those cases are not as severe as the original reported issue, but still we'd have a dangling reference to the latest value. Seems to be a pretty easy error to have in nodejs, as even a `yield*` does have this issue itself xD

Reply:
oh wow, TIL!